aptos-move/framework/aptos-framework/sources/dora_committee.move

```diff
@@ -0,0 +1,1042 @@
+    public entry fun remove_committee(
+        com_store_addr: address,
+        owner_signer: &signer,
+        id: u64
+    ) acquires CommitteeInfoStore, SupraCommitteeEventHandler {
+        // Only the OwnerCap capability can access it
+        let _acquire = &capability::acquire(owner_signer, &OwnerCap {});
+
+        let committee_store = borrow_global_mut<CommitteeInfoStore>(com_store_addr);
+        let (id, committee_info) = simple_map::remove(&mut committee_store.committee_map, &id);
+        // Also remove the node_to_committee_map
+        let (addrs, _) = simple_map::to_vec_pair(committee_info.map);
+        while (vector::length(&addrs) > 0) {
+            let addr = vector::pop_back(&mut addrs);
+            simple_map::remove(&mut committee_store.node_to_committee_map, &addr);
+        };
+        let owner_address = signer::address_of(owner_signer);
+        let event_handler = borrow_global_mut<SupraCommitteeEventHandler>(owner_address);
+        emit_event(
+            &mut event_handler.remove_committee,
+            RemoveCommitteeEvent {
+                committee_id: id,
+                committee_info: committee_info
+            }, )
+    }
+
+    /// Remove the committee in bulk
+    public entry fun remove_committee_bulk(
+        com_store_addr: address,
+        owner_signer: &signer,
+        ids: vector<u64>
+    ) acquires CommitteeInfoStore, SupraCommitteeEventHandler {
+        while (vector::length(&ids) > 0) {
+            let id = vector::pop_back(&mut ids);
+            remove_committee(com_store_addr, owner_signer, id);
+        }
+    }
+
+    /// Upsert the node to the committee
+    public entry fun upsert_committee_member(
+        com_store_addr: address,
+        owner_signer: &signer,
+        id: u64,
+        node_address: address,
+        ip_public_address: vector<u8>,
+        dora_public_key: vector<u8>,
+        network_public_key: vector<u8>,
+        elgamal_pub_key: vector<u8>,
+        network_port: u16,
+        rpc_port: u16,
+    ) acquires CommitteeInfoStore, SupraCommitteeEventHandler {
+        // Only the OwnerCap capability can access it
+        let _acquire = &capability::acquire(owner_signer, &OwnerCap {});
+
+        let committee_store = borrow_global_mut<CommitteeInfoStore>(com_store_addr);
+        let committee = simple_map::borrow_mut(&mut committee_store.committee_map, &id);
+        let dora_node_info = DoraNodeInfo {
+            ip_public_address: copy ip_public_address,
+            dora_public_key: copy dora_public_key,
+            network_public_key: copy network_public_key,
+            elgamal_pub_key: copy elgamal_pub_key,
+            network_port: network_port,
+            rpc_port: rpc_port,
+        };
+        let owner_address = signer::address_of(owner_signer);
+        let event_handler = borrow_global_mut<SupraCommitteeEventHandler>(owner_address);
+        if (!simple_map::contains_key(&committee.map, &node_address)) {
```

Comment:
Can we create common internal functions like `is_node_address_exist()` while returning **bool** value, and use it eveywhere

Reply:
`does_node_exist`

Reply:
@nizam-supraoracles Thanks, addressed in new commit.